### DISCUSSION TITLE
an error occurs when I try to make a transaction

### DISCUSSION THREAD

**Discussion by sasdecompte** [nguyenphuminh/JeChain#83] [General]
Hello
why i have this error -> "ERROR [2023-12-28T00:42:35.518Z] Failed to add one transaction to pool." when I try to make transaction ?
Is there an error in my new transaction object ? ->

```
const transaction = new Transaction("041ce8d242440c5dc46868eb73abc913c9875ebebcf3152070ae146655bfdeebf082ba329fc4a565040873a2cfe56f46dd045153654b48fb1ee794bec04d2105fd", "100", "1", {}, 10);

Transaction.sign(transaction, ec.keyFromPrivate("84f054b3f3183be5dab51d60d34cf6e0783ac9db6671ade83dfbda0f1cc88860"));

axios
.post("http://localhost:5001/sendTransaction", {
params: { transaction: Transaction.serialize(transaction)},
})
.then((response) => {console.log(Response, response.data);})
.catch((error) => {console.error(2ndError, error.response.data);}); 
```

**Comment by nguyenphuminh** [comment]
There could be several reasons to why this would happen:

1. Insufficient balance: Initially, the whole total supply is minted to one account only, so I guess the above failed because you have no coin in your account. To mint the supply into your account, change `FIRST_ACCOUNT` in `src/config.json` to your address.
2. Incorrect nonce: I see you have your nonce set to 10. But if you have not made any transaction before then it should have been 1. To programmatically get the nonce you can also use the `get_nonce` rpc method.
3. Incorrect address: In the past, addresses in JeChain can be whatever, but now it must have the length of 32 bytes (64 characters) and is a valid hex string, more precisely, addresses in JeChain are SHA256 hashes of public keys, you just need to get the SHA256 hash of the address in your code and you are good.

If you have any further problem, feel free to ask!

**Reply by sasdecompte** [reply]
First of all thank you for your reply,
I generated two addresses with keygen.js -> /src/utils/keygen.js

Address n°1, mining the blocks -> 
Address : "60aafa56e838661b90a147d2fe13149ef8e3513a44530c5a7fa37301985ed349"
Public key : "04641218a10e698a21ac9eba4e3b29a30a9dd07bbb8d98136b576bbc19fe6f3187195936550cf4ce8d898b9e9687345ab62351df350f17d777e94e9cbaa9092f87"
Private key: "84f054b3f3183be5dab51d60d34cf6e0783ac9db6671ade83dfbda0f1cc88860"

Address n°2, do nothing
Address : "535f3a402d563320e153d902bcfe178572128033adc7d293760d4dd78c5c53d1"
Public key : "041ce8d242440c5dc46868eb73abc913c9875ebebcf3152070ae146655bfdeebf082ba329fc4a565040873a2cfe56f46dd045153654b48fb1ee794bec04d2105fd"
Private key: "a497a4a79ca256f5855e4b07d155c2a5f3c44d5d68700e58f7c6fc097efc34e1"

When i check the balance of each account i have these responses : 

FIRST_ACCOUNT -> 
Response { success: true, payload: { balance: '100000000202977000000000000' } }

Address n°1 -> 
Response { success: true, payload: { balance: '24357240000000000000' } }

Address n°2 -> 
2ndError {  success: false,  payload: null,  error: { message: 'Invalid request.' }}
(this is normal, this address is empty)

When i get_nonce i have this response ->
Response { success: true, payload: { nonce: 0 } }

I'm trying to send 100 from Address n°1  to Address n°2,
and when I write the transaction, i put this : 

```
const transaction = new Transaction("041ce8d242440c5dc46868eb73abc913c9875ebebcf3152070ae146655bfdeebf082ba329fc4a565040873a2cfe56f46dd045153654b48fb1ee794bec04d2105fd", "100", "1",{}, 0);
Transaction.sign(transaction, ec.keyFromPrivate("84f054b3f3183be5dab51d60d34cf6e0783ac9db6671ade83dfbda0f1cc88860"));

axios
.post("http://localhost:5001/sendTransaction", {
params: { transaction: Transaction.serialize(transaction)},
})
.then((response) => {console.log(Response, response.data);})
.catch((error) => {console.error(2ndError, error.response.data);}); 
```
in the terminal i have this response ->
Response { success: true, payload: { message: 'tx received.' } }

But after i have this response ->
LOG [2024-01-05T01:04:48.673Z] Sent one transaction.
ERROR [2024-01-05T01:04:48.743Z] Failed to add one transaction to pool.

when i tried with "nonce : 1" or changed FIRST_ACCOUNT with Address n°1, I have the same error.

So do you have any idea where the error could come from?
In advance, thank you for your answer

**Reply by nguyenphuminh** [reply]
Oh it's the point 3 that I have made, addresses are not public keys, they are SHA256 hashes of public keys, so I think changing `041ce8d242440c5dc46868eb73abc913c9875ebebcf3152070ae146655bfdeebf082ba329fc4a565040873a2cfe56f46dd045153654b48fb1ee794bec04d2105fd` to `535f3a402d563320e153d902bcfe178572128033adc7d293760d4dd78c5c53d1` would be okay.

**Reply by sasdecompte** [reply]
sorry... I didn't understand this point, thanks for your patience. Now it works. Great!
